src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -555,52 +563,115 @@ def utask_main(uworker_input):
-        deadline, uworker_input.testcase_id, uworker_input.job_type,
-        input_directory, file_list, data, testcase_file_path):
-      return None
+        deadline, testcase, input_directory, file_list, data,
+        testcase_file_path, minimize_task_output):
+      return uworker_msg_pb2.Output(
+          error_type=uworker_msg_pb2.ErrorType.MINIMIZE_DEADLINE_EXCEEDED,
+          minimize_task_output=minimize_task_output)
 
   command = testcase_manager.get_command_line_for_application(
       testcase_file_path, app_args=app_arguments, needs_http=testcase.http_flag)
   last_crash_result = test_runner.last_failing_result
 
   store_minimized_testcase(testcase, input_directory, file_list, data,
-                           testcase_file_path)
+                           testcase_file_path, minimize_task_output)
+
+  minimize_task_output.last_crash_result_dict.clear()
+  minimize_task_output.last_crash_result_dict.update(
+      _extract_crash_result(last_crash_result, command))
 
-  minimize_task_output = uworker_msg_pb2.MinimizeTaskOutput(
-      last_crash_result_dict=_extract_crash_result(last_crash_result, command),
-      flaky_stack=flaky_stack)
   return uworker_msg_pb2.Output(minimize_task_output=minimize_task_output)
 
 
 HANDLED_ERRORS = [
     uworker_msg_pb2.ErrorType.MINIMIZE_SETUP,
     uworker_msg_pb2.ErrorType.TESTCASE_SETUP,
+    uworker_msg_pb2.ErrorType.MINIMIZE_UNREPRODUCIBLE_CRASH,
+    uworker_msg_pb2.ErrorType.MINIMIZE_CRASH_TOO_FLAKY,
+    uworker_msg_pb2.ErrorType.MINIMIZE_DEADLINE_EXCEEDED,
+    uworker_msg_pb2.ErrorType.MINIMIZE_DEADLINE_EXCEEDED_IN_MAIN_FILE_PHASE,
+    uworker_msg_pb2.ErrorType.LIBFUZZER_MINIMIZATION_UNREPRODUCIBLE,
+    uworker_msg_pb2.ErrorType.LIBFUZZER_MINIMIZATION_FAILED,
     uworker_msg_pb2.ErrorType.UNHANDLED
 ]
 
 
+def update_testcase(output: uworker_msg_pb2.Output):
+  """Updates the tescase using the values passed from utask_main. This done at
+  the beginning of utask_postprocess and before error handling is called."""
+  if not output.HasField('minimize_task_output'):
+    return
+
+  minimize_task_output = output.minimize_task_output
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+
+  _update_testcase_memory_tool_options(testcase,
+                                       minimize_task_output.memory_tool_options)
+
+  if minimize_task_output.security_severity_updated:
+    if minimize_task_output.HasField('security_severity'):
+      testcase.security_severity = minimize_task_output.security_severity
+    else:
+      testcase.security_severity = None
+
+  if minimize_task_output.HasField('minimization_phase'):
+    testcase.set_metadata('minimization_phase',
+                          minimize_task_output.minimization_phase)
+
+  if minimize_task_output.flaky_stack:
+    testcase.flaky_stack = minimize_task_output.flaky_stack
+
+  if minimize_task_output.HasField('minimized_arguments'):
```

Comment:
ooo don't like HasField.

Reply:
are there any alternatives here? if not, seems OK to rely on these.

Reply:
I want to only update a testcase field if it was explicitly updated during `utask_main`. I don't see any other way than this for the moment, but I'm open to suggestions.